src/modules/rlm_python/rlm_python.c

```diff
@@ -587,19 +639,33 @@ static unlang_action_t do_python_single(rlm_rcode_t *p_result,
 			rcode = RLM_MODULE_FAIL;
 			goto finish;
 		}
+
 		/* Now have the return value */
 		rcode = PyLong_AsLong(p_tuple_int);
-		/* Reply item tuple */
+
+		/* process updateDict */
+		update_dict = PyTuple_GET_ITEM(p_ret, 1);
+		if (!PyDict_CheckExact(update_dict)) {
+			ERROR("%s - updateDict is not dictionary", funcname);
+			rcode = RLM_MODULE_FAIL;
+			goto finish;
+		}
+
+		mod_vptuple(request->packet, inst, request, &request->request_pairs,
```

Comment:
TBH macros here could shorten this code.  Repetition is opportunity for mistakes